lib/src/group_button_body.dart

```diff
@@ -57,74 +56,44 @@ class GroupButtonBody extends StatefulWidget {
   final List<BoxShadow> unselectedShadow;
   final double? buttonWidth;
   final double? buttonHeigth;
-
   final GroupingType groupingType;
   final MainGroupAlignment mainGroupAlignment;
   final CrossGroupAlignment crossGroupAlignment;
   final GroupRunAlignment groupRunAlignment;
-
   final TextAlign textAlign;
   final EdgeInsets textPadding;
   final AlignmentGeometry? alignment;
   final double? elevation;
+  final GroupButtonController controller;
 
   @override
   _GroupButtonBodyState createState() => _GroupButtonBodyState();
 }
 
 class _GroupButtonBodyState extends State<GroupButtonBody> {
-  int? _selectedIndex;
-  final Map<int, bool> _selectedIndexes = {};
-
   @override
   void initState() {
     super.initState();
-    if (widget.selectedButtons != null && widget.selectedButtons!.isNotEmpty) {
-      // ignore: avoid_function_literals_in_foreach_calls
-      widget.selectedButtons!.forEach((e) {
-        if (!(widget.disabledButtons?.contains(e) ?? false)) {
-          _selectedIndexes[e] = true;
-        }
-      });
-      setState(() {});
-      if (widget.controller != null) {
-        widget.controller?.setSelectedIndexes(widget.selectedButtons!);
-      }
-    }
-    if (widget.selectedButton != null) {
-      if (widget.controller != null) {
-        widget.controller?.setSelectedIndex(widget.selectedButton!);
-      }
-      setState(() => _selectedIndex = widget.selectedButton);
-    }
-    if (widget.controller != null) {
```

Comment:
@Frezyx And how state will react to external changes if widget.controller!.addListener() is removed ? So it is now impossible to use controller for changing widget from outside?